backend/controller/door.go

```diff
@@ -0,0 +1,38 @@
+package controller
+
+import (
+	"api/helpers"
+	"api/models"
+	"encoding/json"
+	"github.com/go-chi/chi/v5"
+	"net/http"
+)
+
+func GetAllDoors(w http.ResponseWriter, _ *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	err, doors := helpers.GetAllTable(w, []models.Door{})
+	if err == nil {
+		helpers.JsonWriter(w, doors)
+	}
+}
+
+func CreateDoor(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	var role models.Door
+	err := json.NewDecoder(r.Body).Decode(&role)
+	// essentially reset if the user inputted anything to common as it should not be editable
+	role.Common = models.Common{}
+	err, user := helpers.CreateNewRecord(w, role, err)
+	if err == nil {
+		helpers.JsonWriter(w, user)
+	}
+}
+
+func DeleteDoor(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	dId := chi.URLParam(r, "doorId")
```

Comment:
should we check null or does deletebyid error report a sufficient error if its null?

Reply:
Wdym? its a url param, im 99% sure that throws a 405 if you just dont pass a param, if you like put /null itll say record not found

Reply:
What if it's formatted incorrectly?

Reply:
works the exact same way as user does, it'll error and won't delete shit and throw the user an eror, I can just add in the parseInt function if we are concerned, but it will error, it just may not be as specific as if I had that fn in there. But this is the same way its done in user so idk if there's been issues using delete for user currently?